stl/inc/yvals.h

```diff
@@ -230,12 +230,10 @@ _STL_DISABLE_CLANG_WARNINGS
 
 #include <use_ansi.h>
 
-#define _WARNING_MESSAGE(NUMBER, MESSAGE) __FILE__ "(" _CRT_STRINGIZE(__LINE__) "): warning " NUMBER ": " MESSAGE
-
 #ifdef _STATIC_CPPLIB
 #ifndef _DISABLE_DEPRECATE_STATIC_CPPLIB
 #ifdef _DLL
-#pragma message(_WARNING_MESSAGE("STL4000", "_STATIC_CPPLIB is deprecated and will be REMOVED."))
+_EMIT_STL_WARNING(4000, "_STATIC_CPPLIB is deprecated and will be REMOVED.");

```

Comment:
This extracts `STL` into the central macros, but I believe that being able to search for `STL4000` in the codebase and find where it's being emitted is important. Although it is slightly redundant, I would prefer to see `STL4000` etc. mentioned in the usage sites.

(For a related example, the compiler's internal sources mention `C4996` etc. when it emits warnings, and I have found this searchability to be very useful. Plain numbers are much harder to search for.)

At a minimum, this makes it easy to search for warning/error number uniqueness.